### ISSUE TITLE
Not Appearing in Nautilus 45.1 on OpenSuse Tumbleweed

### ISSUE THREAD

**Issue by FGYada1984** [costales/folder-color#36]
Hello!

I am on OpenSuse Tumbleweed, Nautilus version 45.1. I installed it through the Yast manager, and installed:

    folder-color-common
    folder-color-common-lang
    nautilus-extension-folder-color

I went through the reboot process, and when opening it through the terminal, no error message appears. However, when clicking on a folder, the options for customization do not appear.

Reading some previous messages here, I tested the command

`G_MESSAGES_DEBUG=all G_DEBUG=all NAUTILUS_DEBUG=all nautilus
`
The first time I ran this command, the only message related to the extension that had appeared was this one:

```
(org.gnome.Nautilus:14841): GLib-DEBUG: 11:38:31.353: setenv()/putenv() are not thread-safe and should not be used after threads are created
/usr/share/nautilus-python/extensions/folder-color.py:19: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '4.0') before import to ensure that the right version gets loaded.
  from gi.repository import Nautilus, Gtk, GObject, Gio, GLib

```

I say it had appeared because, sometime later when I came to write this message, I ran the command again to be able to copy it, but it did not appear anymore, as can be seen below:

```

➜  G_MESSAGES_DEBUG=all G_DEBUG=all NAUTILUS_DEBUG=all nautilus                                                                11/01/23 - 11:47 
(org.gnome.Nautilus:18325): GLib-GIO-DEBUG: 11:47:45.158: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
(org.gnome.Nautilus:18325): GLib-GIO-DEBUG: 11:47:45.160: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ?gio-vfs?
(org.gnome.Nautilus:18325): Tracker-DEBUG: 11:47:45.167: Loading ontologies from database.
(org.gnome.Nautilus:18325): Tracker-DEBUG: 11:47:45.167: Applying ontologies from /usr/share/nautilus/ontology to existing database
(org.gnome.Nautilus:18325): Tracker-DEBUG: 11:47:45.168: Current and DB locales match: 'C'
** Message: 11:47:45.168: Connecting to org.freedesktop.Tracker3.Miner.Files
** (org.gnome.Nautilus:18325): DEBUG: 11:47:45.169: *** Cancel Results Meta requests
(org.gnome.Nautilus:18325): Tracker-DEBUG: 11:47:45.233: Cleaning up stale resource URIs

```

Any idea what it might be?

**Comment by costales** [comment]
Hi,
Could you try with a supported theme, as Numix? Folder Color menu is showed only if the theme supports Folder Color.
Could it be that?
Best regards.

**Comment by FGYada1984** [comment]
Hello,
Thanks for the reply, I already have Papirus which according to the documentation is supported and in fact, a short time ago I was on Nobara 38 which already comes with Papirus installed and it worked normally. The strange thing is that it doesn't give any error messages, only the options don't appear.

**Comment by costales** [comment]
Hi,
I suppose you are using GNOME 4? For your packages I think your packages are not right, because [i don't use folder-color-common anymore](https://launchpad.net/~costales/+archive/ubuntu/folder-color?field.series_filter=mantic).
Could you completly remove all `folder-color` packages and just [copy this file](https://github.com/costales/folder-color/blob/master/nautilus-extension/folder-color.py) (for GTK4) here: `/usr/share/nautilus-python/extensions/folder-color.py`
Restart nautilus with `nautilus -q` and try again.
Best regards.
PS: Folder Color is just a script, the package is for an easy installation, but just copying the file is enough.

**Comment by FGYada1984** [comment]
it worked!
i did exactly what you told me and it worked. probably the official tumbleweed repository is out of date or something. i confess i'm just starting out with opensuse so i'm not sure how everything works. 

thank you very much for your help and work.

**Comment by lusmo** [comment]
Hello, thank you very much for your work on folder-colors! 
Is there any way to use this extension for nautilus in fedora please? Copr repo for Fedora 39 is not working, Fedora 40 is out the door. 
I tried to use the procedure you described, but I was not successful.
I couldn't find any way to get this add-on working. 
I really miss this feature in Fedora. 
Thank you!

**Comment by costales** [comment]
> just [copy this file](https://github.com/costales/folder-color/blob/master/nautilus-extension/folder-color.py) (for GTK4) here: `/usr/share/nautilus-python/extensions/folder-color.py`
> Restart nautilus with `nautilus -q` and try again

Hi, could you try this:

https://github.com/costales/folder-color/issues/36#issuecomment-1791193223

and run nautilus from the Terminal and paste here the result?

Best regards.

**Comment by lusmo** [comment]
I get this:


 ```
File "/usr/share/nautilus-python/extensions/folder-color.py", line 107
    <title>folder-color/nautilus-extension/folder-color.py at master · costales/folder-color · GitHub</title>
                                                                     ^
SyntaxError: invalid character '·' (U+00B7)
```

Thank you!

**Comment by costales** [comment]
> I get this:
> 
> ```
> File "/usr/share/nautilus-python/extensions/folder-color.py", line 107
>    <title>folder-color/nautilus-extension/folder-color.py at master · costales/folder-color · GitHub</title>
>                                                                     ^
> SyntaxError: invalid character '·' (U+00B7)
> ```
> 
> Thank you!

I think you copied a HTML file. Could you try to overwrite the file again with the RAW version?

https://raw.githubusercontent.com/costales/folder-color/master/nautilus-extension/folder-color.py

Best regards.

**Comment by lusmo** [comment]
I'm terribly sorry, I figured it out too and tried to delete my previous post, but I couldn't.
I don't understand how it happened (that's what happens when you do a lot of things at once).
I'm so sorry. I'll fix it...

**Comment by costales** [comment]
Did it work with the raw version? :)

**Comment by lusmo** [comment]
When using the path `/usr/share/nautilus-python/extensions nautilus` would not open at all. No listings. I tried changing the rights but it won't start and offers the option to "wait" or "exit" but neither could be clicked.

I moved the script to` $HOMES/.local/share/nautilus-python/extensions` and changed the user rights.
At first it looked the same, but after a while Nautilus started up.
No listings. I had to change the icons to Numix. Then it was possible to change the color of the icons in the home directory. 
Now Nautilus won't start again and can't even be shut down (not even with the nautilus -q command).
I suspect SELinux, but I have to go home now and I won't be able to do more right now. I'll try to get back to it later.
Thank you very much for your help! 

PS: Just a quick update: After disabling SELinux and choosing Humanity, the nautilus icon starts and it is possible to add emblems, but not to change colors.  I'll check the rights tomorrow (hopefully I'll get around to it). 
Thank you for your help!
